operator/pkg/securedcluster/scanner/auto_sense.go

```diff
@@ -29,6 +46,26 @@ func AutoSenseLocalScannerSupport(ctx context.Context, client ctrlClient.Client,
 	return false, errors.Errorf("invalid spec.scanner.scannerComponent %q", scannerComponent)
 }
 
+func isRunningOnOpenShift(ctx context.Context, client ctrlClient.Client) (bool, error) {
+	log := ctrlLog.FromContext(ctx)
+
+	clusterVersion := &osconfigv1.ClusterVersion{}
```

Comment:
Hm, AFAIK if the scheme does not know the type you can't pass it to an unstructured. At least the helm operator needs a registered type too, otherwise client-go returns an error.
+1 if this works

Reply:
I think the scheme only provides a mapping between the go struct types and k8s kinds. If we use unstructured there is no need for a mapping 🤔

Reply:
Yes, schema registration is redundant for `unstructured` if I understand [this comment](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#pkg-types) correctly . I tested locally without schema but with unstructured and it works. I moved implementation to use `unstructured`. However it's pretty convenient to use predefined type in tests.  Not sure should I adopt tests as well or not